db/types/operations/convert.py

```diff
@@ -18,12 +19,11 @@ def get_db_type_enum_from_id(db_type_id):
         try:
             return MathesarCustomType(db_type_id)
         except ValueError:
+            if db_type_id == UnknownType().id:
+                return UnknownType()
```

Comment:
This change especially, but also the other change in this file might seem funny. The motivation was to get `get_db_type_enum_from_id` and `get_db_type_enum_from_class` to work with unknown types, and it turns out that most of the rest of the code fell into place.